src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -631,10 +631,10 @@ private boolean hasSameNameAsSuperClass(String superClassName, TypeDeclDesc type
      * @param typeDeclWithSameName typeDeclarations which have the same name as the super class
      * @return the nearest class
      */
-    private static TypeDeclDesc getTheNearestClass(String outerTypeDeclName,
+    private static TypeDeclDesc getTheNearestClassOfAnonInnerClass(String outerTypeDeclName,
```

Comment:
Shouldn’t this be `getClosestMatchingTypeDeclaration`?

Reply:
The naming conventions in this check really make this difficult to understand: the type we use for nearly all the heavy lifting is called `TypeDeclDesc`, but we nearly always name the variables with `class`.

Reply:
>getClosestMatchingTypeDeclaration

Ok, Do you fully dislike `OfAnonInner` clarification in name?

Reply:
I don’t know why it matters.

Reply:
It is very specific method with very specific implementation